share/completions/istioctl.fish

```diff
@@ -0,0 +1,176 @@
+# fish completion for istioctl                             -*- shell-script -*-
+
+function __istioctl_debug
+    set -l file "$BASH_COMP_DEBUG_FILE"
+    if test -n "$file"
+        echo "$argv" >> $file
+    end
+end
+
+function __istioctl_perform_completion
+    __istioctl_debug "Starting __istioctl_perform_completion"
+
+    # Extract all args except the last one
+    set -l args (commandline -opc)
+    # Extract the last arg and escape it in case it is a space
+    set -l lastArg (string escape -- (commandline -ct))
+
+    __istioctl_debug "args: $args"
+    __istioctl_debug "last arg: $lastArg"
+
+    set -l requestComp "$args[1] __complete $args[2..-1] $lastArg"
+
+    __istioctl_debug "Calling $requestComp"
+    set -l results (eval $requestComp 2> /dev/null)
+
+    # Some programs may output extra empty lines after the directive.
+    # Let's ignore them or else it will break completion.
+    # Ref: https://github.com/spf13/cobra/issues/1279
+    for line in $results[-1..1]
```

Comment:
This entire block can just be

```fish
set results (string trim -- $results | string match -r '.+')
```

if the issue is empty (or whitespace-only) lines.